transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/AddTransferMethodFragment.java

```diff
@@ -598,50 +598,68 @@ private void setVisibleAndDisableFields() {
         mCreateTransferMethodButton.setBackgroundColor(getResources().getColor(R.color.colorSecondaryDark));
     }
 
-    private boolean performValidation(boolean bypassFocusCheck) {
+    /**
+     * Use this when we want to perform validation on the entire form, typically used when about to submit the form.
+     */
+    private boolean performValidation() {
```

Comment:
This method contains a few issues:

- The variable context doesn't seem to be used

- Cast being done for each call of isWidgetItemValid when the cast has been previously done. You can invoke isWidgetItemValid(widget) instead of isWidgetItemValid((AbstractWidget) v.getTag())

- The implementation is a bit confusing where invoke the isWidgetItemValid in the if and else branches

- The return order is confusing. `return valid && hasWidget` I think it should be hasWidget && isValid 

I think the following implementation would be a little bit clearer. You can swap the usage of  containsInvalidWidget to true and change the logic accordingly, but it's the same idea
```suggestion
    private boolean performValidation() {
        boolean isWidgetValid;
        boolean containsInvalidWidget = false;
        // this is added since some phones triggers the create button but the widgets are not yet initialized
        boolean hasWidget = false;
        for (int i = 0; i < mDynamicContainer.getChildCount(); i++) {
            View v = mDynamicContainer.getChildAt(i);
            if (v.getTag() instanceof AbstractWidget) {
                hasWidget = true;

                AbstractWidget widget = (AbstractWidget) v.getTag();
                WidgetInputState widgetInputState = mWidgetInputStateHashMap.get(widget.getName());
                widgetInputState.setValue(widget.getValue());

                isWidgetValid = isWidgetItemValid(widget);
                if (!isWidgetValid) {
                    containsInvalidWidget = true;
                }
            }
        }
        return hasWidget && !containsInvalidWidget;
    }
```

Reply:
used your code suggestion with a few changes

Reply:
Cool, code looks good! Just some nitpicking... the variable isWidgetValid could be declared inside the loop with the method call isWidgetItemValid and inside the if, you could avoid invoking requireContext().getResources() twice by assigning it to a variable.